File: java/src/main/java/com/penapereira/patterns/simplefactory/SimpleFactory.java
package com.penapereira.patterns.simplefactory;

/** Maps a type code to a concrete product; the single place where products are created. */
public final class SimpleFactory {

    public Product createProduct(String type) {
        return switch (type) {
            case "A" -> new ConcreteProductA();
            case "B" -> new ConcreteProductB();
            default -> throw new IllegalArgumentException("Unknown product type: " + type);
        };
    }
}

File: javascript/src/simple-factory/simple-factory.js
import { ConcreteProductA } from "./concrete-product-a.js";
import { ConcreteProductB } from "./concrete-product-b.js";

/** Maps a type code to a concrete product; the single place where products are created. */
export class SimpleFactory {
  createProduct(type) {
    switch (type) {
      case "A":
        return new ConcreteProductA();
      case "B":
        return new ConcreteProductB();
      default:
        throw new Error(`Unknown product type: ${type}`);
    }
  }
}
